pkg/storage/image.go

```diff
@@ -555,6 +556,13 @@ func (svc *imageService) ResolveNames(imageName string) ([]string, error) {
 	}
 	domain, remainder := splitDockerDomain(imageName)
 	if domain != "" {
+		registry, err := sysregistriesv2.FindRegistry(&types.SystemContext{}, domain)
+		if err != nil {
+			return nil, err
+		}
+		if registry.Blocked {
+			return nil, fmt.Errorf("cannot resolve names from registry %q because it's blocked", domain)
+		}
 		// this means the image is already fully qualified
 		return []string{imageName}, nil
```

Comment:
Shouldn’t the non-qualified search code support the “blocked” flag as well?

Reply:
yup forgot

Reply:
done, basically filtering out the additional registries on startup

Reply:
@mtrmac a feeling inside me grows that the blocked semantics should move into c/image/docker to avoid the pitfalls (and the redundant calls to `FindRegistries(...)` + boilerplate across users). Let's all put our heads together this week at DevConf and check if there are similar cases to make sysregistriesv2 easier to use.